### ISSUE TITLE
docs: Document why overlaps can happen and how to resolve it.

### ISSUE THREAD

**Issue by bwplotka** [thanos-io/thanos#2182]
TL;DR They should never happen in healthy and well configured Thanos system. That's why there is no automatic repair.

This is important to state before we jump into vertical and offline dedups etc cc @metalmatze 

## Some reasons for the compactor seeing overlap:

* Misconfiguraiton of sideca/ruler: Same external labels or no external labels across many block producers
* Running multiple compactors for single block "stream", even for short duration!
* Manually uploading blocks to the bucket.
* Eventually consistent block storage until we fully implement: https://thanos.io/proposals/201901-read-write-operations-bucket.md/ cc @khyatisoneji 

## How to resolve:

Typically you should start with learning HOW it happened.
* Compactor can be blocked for some time, but if it is urgent. Mitigate by removing overlap or better: Backing up somewhere else (you can rename block ULID to non-ulid).
* Who uploaded the block? Search for logs with this ULID across all sidecars/rulers. Check access logs to object storage. Check debug/metas or meta.json of problematic block to see
how blocks looks like and what is the `source`.
* Determine what you misconfigured. If all looks sane and you double-checked everything: Then post an issue on Github, Bugs can happen but we heavily test against such problems.

## Help wanted

Move this to our docs.

**Comment by bwplotka** [comment]
Fix in progress: https://github.com/thanos-io/thanos/pull/2191/files
